scripts/capa2sarif.py

```diff
@@ -0,0 +1,343 @@
+optional arguments:
+  -h, --help            show this help message and exit
+  --version             show program's version number and exit
+  -t TAG, --tag TAG     filter on rule meta field values (ruleid)
+"""
+import logging
+import sys
+import json
+import argparse
+from pathlib import Path
+
+from sarif_om import Tool, SarifLog, Run, ToolComponent
+from jschema_to_python.to_json import to_json
+
+from capa.version import __version__
+
+from typing import Optional, List
+
+
+logger = logging.getLogger('capa2sarif')
+
+
+def _parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="Capa to SARIF analysis file"
+    )
+
+    # Positional argument
+    parser.add_argument(
+        'capa_output',
+        help='Path to capa JSON output file'
+    )
+
+    # Optional arguments
+    parser.add_argument(
+        '-g', '--ghidra-compat',
+        action="store_true",
+        help='Compatibility for Ghidra 11.0.X'
+    )
+    parser.add_argument(
+        '-r', '--radare-compat',
+        action="store_true",
+        help='Compatibility for Radare r2sarif plugin v2.0'
+    )
+    parser.add_argument(
+        '-t', '--tag',
+        help='Filter on rule meta field values (ruleid)'
+    )
+    parser.add_argument(
+        '--version',
+        action='version',
+        version=f'%(prog)s {__version__}'
+    )
+
+    return parser.parse_args()
+
+
+def main() -> int:
+    logging.basicConfig(level=logging.INFO)
+    logging.getLogger().setLevel(logging.INFO)
+
+    args = _parse_args()
+
+    try:
+        with Path(args.capa_output).open() as capa_output:
+            json_data = json.load(capa_output)
+    except ValueError:
+        logger.error("Input data was not valid JSON, input should be a capa json output file.")
+        return -1
+    except json.JSONDecodeError:
+        # An exception has occured
+        logger.error("Input data was not valid JSON, input should be a capa json output file.")
+        return -2
+
+    # Marshall json into Sarif
+    # Create baseline sarif structure to be populated from json data
+    sarif_structure: Optional[dict] = _sarif_boilerplate(json_data["meta"], json_data["rules"])
+    if sarif_structure is None:
+        logger.errort("An Error has occured creating default sarif structure.")
+        return -3
+
+    _populate_artifact(sarif_structure, json_data["meta"])
+    _populate_invoations(sarif_structure, json_data["meta"])
+    _populate_results(sarif_structure, json_data["rules"], args.ghidra_compat)
+
+    if args.ghidra_compat:
+        # Ghidra can't handle this structure as of 11.0.x
+        if 'invocations' in sarif_structure['runs'][0]:
+            del sarif_structure['runs'][0]['invocations']
+
+        # artifacts must include a description as well with a text field.
+        if 'artifacts' in sarif_structure['runs'][0]:
+            sarif_structure['runs'][0]['artifacts'][0]['description'] = {'text': 'placeholder'}
```

Comment:
single quotes vs double quotes are inconsistent here. would you please run this script through `black`?